android/app/src/main/java/org/fossasia/openevent/viewholders/LocationViewHolder.java

```diff
@@ -0,0 +1,58 @@
+package org.fossasia.openevent.viewholders;
+
+import android.content.Context;
+import android.content.Intent;
+import android.support.v7.widget.RecyclerView;
+import android.view.View;
+import android.widget.TextView;
+
+import org.fossasia.openevent.R;
+import org.fossasia.openevent.activities.LocationActivity;
+import org.fossasia.openevent.data.Microlocation;
+import org.fossasia.openevent.utils.ConstantStrings;
+
+import java.text.MessageFormat;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+
+/**
+ * Created by Shreyas on 5/17/2017.
+ */
+
+public class LocationViewHolder extends RecyclerView.ViewHolder {
+
+    @BindView(R.id.location_name)
+    TextView locationName;
+
+    @BindView(R.id.location_floor)
+    TextView locationFloor;
+
+    private Microlocation location;
+    private Context context;
+
+
+    public void setLocation(Microlocation location) {
+        this.location = location;
+    }
+
+    public LocationViewHolder(View itemView, Context context) {
+        super(itemView);
+        this.context = context;
+        ButterKnife.bind(this, itemView);
+    }
+
+    public void bindHolder(){
+        locationName.setText(location.getName());
+        locationFloor.setText(MessageFormat.format("{0}{1}",
+                itemView.getResources().getString(R.string.fmt_floor),
+                location.getFloor()));
+
+        itemView.setOnClickListener(v -> {
```

Comment:
Attach the click listener in the constructor itself. This way, a new object won't be created on each bind call